MapboxMobileEventsTests/MMEAPIClientTests.mm

```diff
@@ -43,6 +47,136 @@ - (void)setupUserAgent;
         apiClient.baseURL should equal([NSURL URLWithString:MMEAPIClientBaseURL]);
     });
     
+    describe(@"- URLSession:didReceiveChallenge:completionHandler:", ^{
+        context(@"when challenge is evaluated by TrustKit", ^{
+            __block bool isMainThread;
+            
+            beforeEach(^{
+                id<CedarDouble> delegateFake = fake_for(@protocol(NSURLAuthenticationChallengeSender));
+                
+                NSURLSessionConfiguration *sessionConfig = [NSURLSessionConfiguration defaultSessionConfiguration];
+                MMENSURLSessionWrapper *sessionWrapper = [[MMENSURLSessionWrapper alloc] init];
+                
+                apiClient.sessionWrapper = sessionWrapper;
+                spy_on(apiClient.sessionWrapper);
+                
+                __block bool completionBlockCalled = false;
+                
+                NSURLSession *urlSession = [NSURLSession sessionWithConfiguration:sessionConfig delegate:sessionWrapper delegateQueue:nil];
+                NSURLProtectionSpace *space = [[NSURLProtectionSpace alloc] initWithHost:@"hostname" port:0 protocol:nil realm:nil authenticationMethod:NSURLAuthenticationMethodServerTrust];
+                NSURLAuthenticationChallenge *challenge = [[NSURLAuthenticationChallenge alloc] initWithProtectionSpace:space proposedCredential:nil previousFailureCount:0 failureResponse:nil error:nil sender:((DelegateTestClass *)delegateFake)];
+                
+                [sessionWrapper URLSession:urlSession didReceiveChallenge:challenge completionHandler:^(NSURLSessionAuthChallengeDisposition, NSURLCredential * _Nullable) {
+                    isMainThread = [NSThread isMainThread];
+                    completionBlockCalled = true;
+                }];
+                
+                while (!completionBlockCalled) {
+                    NSDate *futureDate = [NSDate dateWithTimeIntervalSinceNow:0.1];
+                    [[NSRunLoop currentRunLoop] runUntilDate:futureDate];
+                }
+            });
+            
+            it(@"should be on background queue", ^{
+                isMainThread should be_falsy;
```

Comment:
The completion block should always be called on the main thread, so is this failing?

Reply:
i.e. I think this `should be_truthy`